### PR TITLE
Better logging around context handling in the HeadTracker

### PR DESCRIPTION
**Pull request by j16r**

### CODE REVIEW

File: core/services/head_tracker.go [hunk 1]
```diff
@@ -267,15 +269,21 @@ func (ht *HeadTracker) listenForNewHeads() {
 		logger.WarnIf(errors.Wrap(err, "HeadTracker failed when unsubscribe from head"))
 	}()
 
+	ctx, cancel := utils.ContextFromChan(ht.done)
+	defer cancel()
+
 	for {
 		if !ht.subscribe() {
-			return
+			break
```
Comment (samsondav) [inline core/services/head_tracker.go:277]: Why change this, doesn't it do the same thing?

File: core/store/orm/log_wrapper.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ func (o *ormLogWrapper) Error(ctx context.Context, s string, i ...interface{}) {
 func (o *ormLogWrapper) Trace(ctx context.Context, begin time.Time, fc func() (string, int64), err error) {
 	elapsed := time.Since(begin)
 	switch {
+	case ctx.Err() != nil:
```
Comment (samsondav) [inline core/store/orm/log_wrapper.go:47]: 👍  👍

**Review by samsondav — APPROVED** [review]
Thanks for fixing this JB

File: core/services/head_tracker.go (hunk 1 above)
Comment (j16r) [reply core/services/head_tracker.go:277]: personal preferences, feels more local

**Review by samsondav — APPROVED** [review]

**Pull request merged**